src/cpu/aarch64/jit_uni_pool_kernel.cpp

```diff
@@ -0,0 +1,1130 @@
+
+        dup(vmm_tmp.s, ZRegS(xmm_tmp.getIdx())[0]);
+
+        add(vmm_k_offset, vmm_k_offset, vmm_tmp.s);
+        subs(ki, ki, 1);
+        mov_imm(x_tmp_0, 0);
+        cmp(ki, x_tmp_0);
+        b(GT, kd_label);
+        ldr(reg_output, post_ptr(X_TRANSLATOR_STACK, 8));
+
+        ldr(reg_input, post_ptr(X_TRANSLATOR_STACK, 8));
+    }
+}
+
+template <cpu_isa_t isa>
+void jit_uni_pool_kernel<isa>::zero_diff_src(
+        int ur_bc, bool with_c_tail_proccessing) {
+    const int c_off = (jpp.tag_kind == jit_memory_tag_kind_t::nspc)
+            ? jpp.c
+            : jpp.c_block;
+
+    Label l_skip, l_ih_loop, l_id_loop;
+
+    auto is_tail_processing = [&](int bc) {
+        return with_c_tail_proccessing && bc == (ur_bc - 1);
+    };
+
+    add_imm(x_tmp_addr, reg_param, GET_OFF(zero_id), x_tmp_0);
+    ldr(reg_zero_id, ptr(x_tmp_addr));
+
+    mov_imm(x_tmp_0, 0);
+    cmp(reg_zero_id, x_tmp_0);
+
+    b(EQ, l_skip);
+
+    add_imm(x_tmp_addr, reg_param, GET_OFF(zero_ih), x_tmp_0);
+    ldr(reg_zero_ih, ptr(x_tmp_addr));
+
+    mov_imm(x_tmp_0, 0);
+    cmp(reg_zero_ih, x_tmp_0);
+
+    b(EQ, l_skip);
+
+    add_imm(x_tmp_addr, reg_param, GET_OFF(zero_ptr), x_tmp_0);
+    ldr(reg_zero_ptr, ptr(x_tmp_addr));
+
+    TReg vzero = vmm_tmp;
+    eor(vzero.d, vzero.d, vzero.d);
+
+    const int width_size = jpp.iw * c_off * jpp.dt_size;
+
+    auto aux_reg_zero_ptr = tmp_gpr;
+
+    L(l_id_loop);
+    {
+        mov(aux_reg_zero_ptr, reg_zero_ptr);
+        mov(aux_reg_zero_ih, reg_zero_ih);
+        L(l_ih_loop);
+        {
+            const int step = c_off * jpp.dt_size;
+
+            // TODO: maybe a big code generated here
+            for_(int i = 0; i < width_size; i += step)
+            for (int bci = 0; bci < ur_bc; bci++) {
+                const int offs = i + bci * jpp.c_block * jpp.dt_size;
+                store(vzero.getIdx(), xreg_zero_ptr, offs,
+                        is_tail_processing(bci));
+            }
+            add_imm(reg_zero_ptr, reg_zero_ptr, width_size, x_tmp_0);
+            subs(aux_reg_zero_ih, aux_reg_zero_ih, 1);
+            b(NE, l_ih_loop);
+        }
+        mov(reg_zero_ptr, aux_reg_zero_ptr);
+        add_imm(reg_zero_ptr, reg_zero_ptr, (width_size * jpp.ih), x_tmp_0);
+        subs(reg_zero_id, reg_zero_id, 1);
+        b(NE, l_id_loop);
+    }
+
+    L(l_skip);
+}
+
+template <cpu_isa_t isa>
+void jit_uni_pool_kernel<isa>::generate() {
+
+    this->preamble();
+
+    Label idx_table;
+
+    int ow = jpp.ow;
+    int iw = jpp.iw;
+    int kw = jpp.kw;
+    int kh = jpp.kh;
+    int c_block = jpp.c_block;
+    int stride_w = jpp.stride_w;
+    int l_pad = jpp.l_pad;
+    const int c_off
+            = (jpp.tag_kind == jit_memory_tag_kind_t::nspc) ? jpp.c : c_block;
+
+    int vlen = cpu_isa_traits<isa>::vlen;
+
+#if defined(_WIN32)
+    // Always mimic the Unix ABI (see the note about maskmovdqu in the header
+    // file).
+    xor_(rdi, rcx);
+    xor_(rcx, rdi);
+    xor_(rdi, rcx);
```

Comment:
I was wondering if above three instructions are not specific for `x64` ISA.

Reply:
Thanks for your reviews. I'll delete it.